src/main/java/org/tikv/common/region/RegionStoreClient.java

```diff
@@ -298,7 +298,16 @@ public List<KvPair> batchGet(BackOffer backOffer, List<ByteString> keys, long ve
             forWrite);
     BatchGetResponse resp =
         callWithRetry(backOffer, TikvGrpc.getKvBatchGetMethod(), request, handler);
-    return handleBatchGetResponse(backOffer, resp, version);
+    try {
+      return handleBatchGetResponse(backOffer, resp, version);
+    } catch (TiKVException e) {
+      if ("locks not resolved, retry".equals(e.getMessage())) {
```

Comment:
We should not retry depending on the error message. could we retry in `handleBatchGetResponse`?

Reply:
handleBatchGetResponse may need to add a parameter `keys`